extensions/amp-base-carousel/1.0/base-carousel.js

```diff
@@ -353,7 +354,12 @@ function BaseCarouselWithRef(
           and next viewport.
         */}
         {childrenArray.map((child, index) =>
-          Math.abs(index - currentSlide) < visibleCount * 3 || mixedLength ? (
+          Math.min(
+            // Distance from currentSlide.
+            Math.abs(index - currentSlide),
+            // Account for wraparound when looping.
+            loop ? mod(length + currentSlide - index, length) : length
+          ) < Math.ceil(visibleCount * 3) || mixedLength ? (
             <WithAmpContext
```

Comment:
Is this the same as:

```
(loop ? mod(length + currentSlide - index, length) : Math.abs(index - currentSlide)) < Math.ceil ...
```

maybe less readable though

Reply:
Not exactly, since we take `Math.min()` But agree, it is not very readable as is.